### PR TITLE
Update libtorch to v1.7

### PR DESCRIPTION
**Pull request by KDr2**

### CODE REVIEW

**Comment by KDr2** [comment]
@yebai Could we merge this PR first, then I will start to separate the binary providing things to a jll package.

File: .github/workflows/BuildDylib-MacOS.yml [hunk 1]
```diff
@@ -17,11 +17,11 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        julia-version: [1.2]
+        julia-version: [1.5]
```
Comment (yebai) [inline .github/workflows/BuildDylib-MacOS.yml:20]:
```suggestion
        julia-version: [1]
```

File: .github/workflows/UnitTest.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        julia-version: [1.0.5, 1.2.0, 1.3]
+        julia-version: [1.2.0, 1.3]
```
Comment (yebai) [inline .github/workflows/UnitTest.yml:16]:
```suggestion
        julia-version: [1, 1.5]
```

File: deps/build_tarballs.jl [hunk 3]
```diff
@@ -31,7 +31,7 @@ function get_commit_id()
 end
 
 sources = [
-    "https://github.com/TuringLang/ThArrays.jl.git" => get_commit_id(),
+    GitSource("https://github.com/TuringLang/ThArrays.jl.git", get_commit_id()),
```
Comment (yebai) [inline deps/build_tarballs.jl:34]:
```suggestion
    GitSource("https://github.com/data-efficient-ml/ThArrays.jl.git", get_commit_id()),
```

File: .github/workflows/UnitTest.yml (hunk 2 above)
Comment (KDr2) [reply .github/workflows/UnitTest.yml]: All updated.

**Pull request merged**